libs/api/domains/icelandic-names-registry/src/lib/models/icelandicName.model.ts

```diff
@@ -25,19 +25,19 @@ export class IcelandicName implements TIcelandicName {
   @IsEnum(StatusType)
   status!: StatusType | null
 
-  @Field({ nullable: true })
+  @Field(() => Boolean, { nullable: true })
   @IsBoolean()
   visible!: boolean | null
```

Comment:
Shouldn't these be:
```typescript
visible?: boolean
```

(same goes for other fields)

Reply:
In strict TypeScript, the Field decorator can not reflect union types:

```
// BAD
@Field({ nullable: true })
description: string | undefined

@Field({ nullable: true })
description!: string | null

@Field({ nullable: true })
description?: string | null

// GOOD
@Field({ nullable: true })
description?: string

@Field(() => String, { nullable: true })
description: string | undefined

@Field(() => String, { nullable: true })
description!: string | null

@Field(() => String, { nullable: true })
description?: string | null
```

I preferred the first "GOOD" option and tried to make it consistent like that, but that caused issues when casting these GraphQL types back and forth with other types.

These other types are usually from client libraries and often based on OpenAPI schemas which sometimes marks properties as nullable (null) and sometimes as not required (undefined). Depending on the flow of data, this causes typescript errors if they don't match correctly. 

So rather than chasing down too many rabbit holes (some of these types come from external OpenAPI schemas), I just fixed the remaining issues by specifying the type in the `@Field` decorator.